crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -97,7 +97,20 @@ pub struct Lexer<'a> {
 
 #[allow(clippy::unused_self)]
 impl<'a> Lexer<'a> {
-    pub fn new(allocator: &'a Allocator, source_text: &'a str, source_type: SourceType) -> Self {
+    /// Create new `Lexer`.
+    ///
+    /// # SAFETY
+    /// Caller must not allow more than one `Lexer` to exist at any one time on a thread.
+    /// In practice, this is easy to satisfy. Only means that, apart from `ParserImpl::new`,
+    /// no method of `ParserImpl` or `Lexer` can call this and create another `Lexer`.
+    #[allow(clippy::missing_safety_doc)] // Clippy is wrong!
```

Comment:
I thought the comment need to start with `SAFETY:`

Reply:
I've been able to remove the unsafe, and hence remove these annoying clippy attrs in refactored version.

Clippy is happy with `# Safety` but refuses to recognise `# SAFETY`. Clippy is an anti-capitalist!